### PR TITLE
[flang] Lower selected_logical_kind to its runtime call

### PR DESCRIPTION
**Pull request by clementval**
Runtime support has been added in #89691. This patch adds lowering in a similar way than `selected_int_kind` and `selected_real_kind`.

### CODE REVIEW

File: flang/lib/Optimizer/Builder/IntrinsicCall.cpp [hunk 1]
```diff
@@ -553,6 +553,10 @@ static constexpr IntrinsicHandler handlers[]{
      &I::genSelectedIntKind,
      {{{"scalar", asAddr}}},
      /*isElemental=*/false},
+    {"selected_logical_kind",
+     &I::genSelectedLogicalKind,
+     {{{"scalar", asAddr}}},
```
Comment (clementval) [inline flang/lib/Optimizer/Builder/IntrinsicCall.cpp:558]:
```suggestion
     {{{"bits", asAddr}}},
```

File: flang/test/Lower/Intrinsics/selected_logical_kind.f90 [hunk 2]
```diff
@@ -0,0 +1,72 @@
+! REQUIRES: shell
```
Comment (vzakhari) [inline flang/test/Lower/Intrinsics/selected_logical_kind.f90:1]: nit: is it really needed?

**Review by vzakhari — APPROVED** [review]
LGTM

File: flang/test/Lower/Intrinsics/selected_logical_kind.f90 (hunk 2 above)
Comment (clementval) [reply flang/test/Lower/Intrinsics/selected_logical_kind.f90:1]: Not sure. I copied it from the selected_int_kind test. I think it was added a while back for windows but I guess it is not necessary here. Let me remove it.

Comment (clementval) [inline flang/test/Lower/Intrinsics/selected_logical_kind.f90:1]:
```suggestion
```

**Pull request merged**